src/CLICommand/common_context.ts

```diff
@@ -0,0 +1,89 @@
+import { DriveKey } from '../arfsdao';
+import { JWKWallet, Wallet, WalletDAO } from '../wallet_new';
+import { ParameterName } from './parameter';
+import * as fs from 'fs';
+import { JWKInterface } from 'ardrive-core-js';
+import {
+	DriveAddressParameter,
+	DriveKeyParameter,
+	DrivePasswordParameter,
+	SeedPhraseParameter,
+	WalletFileParameter
+} from '../parameter_declarations';
+import Arweave from 'arweave';
+
+/**
+ * @type {CommonContext}
+ * A class representing the context of the parameters
+ */
+export class CommonContext {
+	private readonly walletDao: WalletDAO;
+
+	/**
+	 * @returns {CommonContext}
+	 * @param {any} options The object containing the parameterName: value mapping
+	 * @param {Arweave} arweave The arweave instance
+	 * An immutable instance of CommonContext holding the parsed values of the parameters
+	 */
+	constructor(private readonly options: any, private readonly arweave: Arweave) {
+		this.walletDao = new WalletDAO(this.arweave);
+	}
+
+	/**
+	 * @returns {Promise<boolean>}
+	 * Returns true when a drive password, drive key or wallet file is provided
+	 */
+	public async getIsPrivate(): Promise<boolean> {
+		return !!(this.password || this.driveKey || (await this.getWallet().catch(() => false)));
+	}
+
+	/**
+	 * @returns {Promise<Wallet>}
+	 * Will return a wallet instance created from the seed phrase or the walletFile
+	 */
+	public async getWallet(): Promise<Wallet> {
+		const walletFile = this.getParameterValue(WalletFileParameter);
+		const seedPhrase = this.getParameterValue(SeedPhraseParameter);
+		if (walletFile) {
+			const walletFileData = fs.readFileSync(walletFile, { encoding: 'utf8', flag: 'r' });
+			const walletJSON = JSON.parse(walletFileData);
+			const walletJWK: JWKInterface = walletJSON as JWKInterface;
+			return new JWKWallet(walletJWK);
+		} else if (seedPhrase) {
+			return await this.walletDao.generateJWKWallet(seedPhrase);
+		}
+		throw new Error('No wallet file neither seed phrase provided!');
+	}
+
+	/**
+	 * @type {string | undefined}
+	 * A getter for the drive password
+	 */
+	public get password(): string | undefined {
```

Comment:
Do we need to make another getter here for each parameter we declare in parameter_declarations?

Reply:
Not really, it's pointless to have those as we have one generic `getParameterValue()`
Removed useless code: bd5eda138d074f05d03c853bb4b808a5f5f5df2d